internal/k8s/entity.go

```diff
@@ -118,27 +101,14 @@ func (e K8sEntity) GVK() schema.GroupVersionKind {
 // Clean up internal bookkeeping fields. See
 // https://github.com/kubernetes/kubernetes/issues/90066
 func (e K8sEntity) Clean() {
-	m := e.meta()
-	m.SetManagedFields(nil)
+	e.Meta().SetManagedFields(nil)
 
-	annotations := m.GetAnnotations()
+	annotations := e.Meta().GetAnnotations()
 	if len(annotations) != 0 {
 		delete(annotations, "kubectl.kubernetes.io/last-applied-configuration")
 	}
 }
 
-func (e K8sEntity) meta() ObjectMeta {
-	if unstruct := e.maybeUnstructuredMeta(); unstruct != nil {
-		return unstruct
-	}
-
-	if structured, _ := e.maybeStructuredMeta(); structured != nil {
-		return structured
-	}
-
-	return emptyMeta{}
-}
-
 func (e K8sEntity) maybeUnstructuredMeta() *unstructured.Unstructured {
```

Comment:
In theory, should this method (+ `maybeStructuredMeta`) go away? It looks like it's now only* used by `SetUID` - should that also be using `e.Meta().SetUID()`?

\* I do see it also gets used in a CRD test but I think that's easily in-lineable

Reply:
ooh good point. removed!